packages/astro/src/astro-components/interactive/OrganizationList.astro

```diff
@@ -0,0 +1,36 @@
+---
+import type { OrganizationListProps } from "@clerk/types";
+type Props = OrganizationListProps
+
+import { customAlphabet, urlAlphabet } from "nanoid";
+
+const safeId = customAlphabet(urlAlphabet, 10)();
+
+const props = {
+  ...Astro.props,
+}
+---
+
+<div id={`clerk-organization-list-${safeId}`}></div>
+
+<script is:inline define:vars={{ props: Astro.props, safeId }}>
+  const setOrCreatePropMap = ({ category, id, props }) => {
+    if (!window.__astro_clerk_component_props) {
+      window.__astro_clerk_component_props = new Map();
+    }
+
+    if (!window.__astro_clerk_component_props.has(category)) {
+      const _ = new Map();
+      _.set(id, props);
+      window.__astro_clerk_component_props.set(category, _);
+    }
+
+    window.__astro_clerk_component_props.get(category)?.set(id, props);
+  };
+
+  setOrCreatePropMap({
+    category: "organization-list",
+    id: `clerk-organization-list-${safeId}`,
+    props,
+  });
```

Comment:
Can you explain what's happening here? Maybe it's my unfamiliarity with Astro, but it's not immediately clear what we're doing with this inline script. Alternatively, is it possible to consolidate this somehow into a single source of truth? Or is there some limitation that requires us to duplicate it in all of the interactive component definitions?

Reply:
@wobsoriano seems to have a way in order to address the multiple duplicates. I remember myself, hitting a limitation but we will revisit this one.

I will add a comment in this file that describes what is happening here.